src/model.ts

```diff
@@ -277,30 +278,36 @@ export class Model {
 	 */
 	prepare(extend: () => void): void {
 		// Create a model initialization scope
-		if (!this._ready) {
+		if (!this._readyCallbacks) {
+			this._readyProcessing = false;
 			// Create an array to track model initialization callbacks
-			Object.defineProperty(this, "_ready", { enumerable: false, configurable: true, writable: true, value: [] });
+			Object.defineProperty(this, "_readyCallbacks", { enumerable: false, configurable: true, writable: true, value: [] });
 
 			// Extend the model
 			extend();
 
 			// Complete pending model initialization steps
-			this._ready.forEach(init => init());
-			delete this._ready;
+			this._readyProcessing = true;
+			this._readyCallbacks.forEach(init => init());
+			this._readyProcessing = false;
+			delete this._readyCallbacks;
 		}
 
 		// Leverage the current model initialization scope
 		else
 			extend();
 	}
 
-	ready(init: () => void): void {
-		if (this._ready) {
-			this._ready.push(init);
-		}
-		else {
-			init();
-		}
+	/**
```

Comment:
Added a new parameter which allows a given callback to be invoked immediately if the queue of "ready callbacks" is already being processed.